Upgrading the cobra version to 1.6.1 causes kctrl tests to fail

**Bug description**
The dependency [github.com/spf13/cobra](http://github.com/spf13/cobra) has a newer version available `v1.6.1`. This was supposed to be upgraded as part of the CAPA version bump in #3874. The bump was causing the kctrl tests to fail with the following error:
```bash
$ go test -timeout 30s -coverprofile=/var/folders/w9/fvjgk13x40s56gvtzqp3q7kh0000gq/T/vscode-golKDwiN/go-code-cover github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/package/kctrl

--- FAIL: TestKctrlInvoke (0.00s)
    --- FAIL: TestKctrlInvoke/Invoke (0.00s)
        --- FAIL: TestKctrlInvoke/Invoke/add_package_commands_from_kctrl (0.00s)
panic: Expected to find available commands section in spf13/cobra default usage template [recovered]
	panic: Expected to find available commands section in spf13/cobra default usage template

goroutine 132 [running]:
testing.tRunner.func1.2({0x2391340, 0x2907540})
	/usr/local/Cellar/go@1.18/1.18.5/libexec/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/Cellar/go@1.18/1.18.5/libexec/src/testing/testing.go:1392 +0x39f
panic({0x2391340, 0x2907540})
	/usr/local/Cellar/go@1.18/1.18.5/libexec/src/runtime/panic.go:838 +0x207
github.com/cppforlife/cobrautil.HelpSectionsUsageTemplate({0xc000090d20, 0x2, 0x269e88d?})
	/Users/muchhals/go/pkg/mod/github.com/cppforlife/cobrautil@v0.0.0-20220907150944-da5ee3a6ab1f/help_sections.go:42 +0x25c
github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd.AddPackageCommands(0xc00071b400, 0x3b902b0?, {{0x29252f8?, 0xc0004f9290?}, {0x2924220?, 0xc0004f92c0?}}, {{0x26960f7, 0x5}, 0x1, 0x0, ...})
	/Users/muchhals/go/pkg/mod/github.com/vmware-tanzu/carvel-kapp-controller/cli@v0.0.0-20221026213745-c834810997d4/pkg/kctrl/cmd/kctrl.go:180 +0x267
github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd.AttachKctrlPackageCommandTree(0x235d540?, 0xc00003c240, {{0x26960f7?, 0x228ce0e?}, 0x40?, 0xd5?, 0x35?})
	/Users/muchhals/go/pkg/mod/github.com/vmware-tanzu/carvel-kapp-controller/cli@v0.0.0-20221026213745-c834810997d4/pkg/kctrl/cmd/kctrl.go:234 +0x254
github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/package/kctrl.Invoke(0xc00019a988)
	/Users/muchhals/workspace/tkg/tanzu-framework/cmd/cli/plugin/package/kctrl/kctrl.go:21 +0x2ae
github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/package/kctrl.TestKctrlInvoke.func1.1(0x0?)
	/Users/muchhals/workspace/tkg/tanzu-framework/cmd/cli/plugin/package/kctrl/kctrl_test.go:27 +0x13c
testing.tRunner(0xc000783d40, 0x2790cb8)
	/usr/local/Cellar/go@1.18/1.18.5/libexec/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/local/Cellar/go@1.18/1.18.5/libexec/src/testing/testing.go:1486 +0x35f
FAIL	github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/package/kctrl	0.444s
FAIL
```

Due to this, we pinned back the version used by `cmd/cli/plugin/package` to v1.5.0 for the time being. This issue talks about addressing the test failure and upgrading the cobra version to the latest available.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**
We should be able to safely upgrade the cobra dependency version.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

--- a/cmd/cli/plugin/package/go.mod
+++ b/cmd/cli/plugin/package/go.mod
@@ -29,7 +29,7 @@ require (
 	github.com/spf13/cobra v1.6.1
 	github.com/stretchr/testify v1.8.1
 	github.com/vmware-tanzu/carvel-kapp-controller v0.39.0
-	github.com/vmware-tanzu/carvel-kapp-controller/cli v0.0.0-20221026213745-c834810997d4
+	github.com/vmware-tanzu/carvel-kapp-controller/cli v0.0.0-20221214062743-6b0843a3db36
 	github.com/vmware-tanzu/tanzu-framework/capabilities/client v0.0.0-00010101000000-000000000000
 	github.com/vmware-tanzu/tanzu-framework/cli/runtime v0.0.0-00010101000000-000000000000
 	github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/secret v0.0.0-00010101000000-000000000000
@@ -88,7 +88,7 @@ require (
 	github.com/containerd/stargz-snapshotter/estargz v0.11.4 // indirect
 	github.com/coredns/caddy v1.1.1 // indirect
 	github.com/coredns/corefile-migration v1.0.18 // indirect
-	github.com/cppforlife/cobrautil v0.0.0-20220907150944-da5ee3a6ab1f // indirect
+	github.com/cppforlife/cobrautil v0.0.0-20221130162803-acdfead391ef // indirect
 	github.com/cppforlife/color v1.9.1-0.20200716202919-6706ac40b835 // indirect
 	github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect

